udata/core/post/views.py

```diff
@@ -28,17 +28,32 @@ class ProtectedPostView(PostView):
 
 
 @blueprint.route('/', endpoint='list')
-def list_posts():
-    return theme.render('post/list.html',
-                        posts=Post.objects.order_by('-created_at'))
+class PostListView(ListView):
+    model = Post
+    template_name = 'post/list.html'
+    context_name = 'posts'
+
+    @property
+    def default_page_size(self):
+        return current_app.config['POST_DEFAULT_PAGINATION']
+
+    def get_queryset(self):
+        return Post.objects(private=False).order_by('-created_at').paginate(self.page, self.page_size)
 
 
 @blueprint.route('/<post:post>/')
 def show(post):
-    return theme.render('post/display.html', post=post)
+    others = Post.objects(id__ne=post.id, private=False)
+    older = others(created_at__lt=post.created_at).order_by('-created_at')
+    newer = others(created_at__gt=post.created_at).order_by('created_at')
+    return theme.render('post/display.html',
+                        post=post,
+                        previous_post=older.first(),
+                        next_post=newer.first())
 
 
 @sitemap.register_generator
 def sitemap_urls():
+    yield 'posts.list_redirect', {}, None, "weekly", 0.6
```

Comment:
Isn't the code below unreachable?

Reply:
I don't think so. After this `yield` statement, I believe the interperter is processing the for loop.
What do you think prevent it ?

Reply:
Well, I always thought of `yield` as a kind of `return` but after a quick test I realize I was wrong ;-)